sycl/source/detail/kernel_compiler/kernel_compiler_sycl.cpp

```diff
@@ -0,0 +1,263 @@
+//==-- kernel_compiler_opencl.cpp  OpenCL kernel compilation support       -==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "kernel_compiler_sycl.hpp"
+#include <sycl/exception.hpp> // make_error_code
+
+#if __GNUC__ && __GNUC__ < 8
+
+// std::filesystem is not availalbe for GCC < 8
+// and much of the  cross-platform file handling code depends upon it.
+// Given that this extension is experimental and that the file
+// handling aspects are most likely temporary, it makes sense to
+// simply not support GCC<8.
+
+namespace sycl {
+inline namespace _V1 {
+namespace ext::oneapi::experimental {
+namespace detail {
+
+bool SYCL_Compilation_Available() { return false; }
+
+spirv_vec_t
+SYCL_to_SPIRV(const std::string &SYCLSource, include_pairs_t IncludePairs,
+              const std::vector<std::string> &UserArgs, std::string *LogPtr,
+              const std::vector<std::string> &RegisteredKernelNames) {
+  throw sycl::exception(sycl::errc::build,
+                        "kernel_compiler does not supprot GCC<8");
+}
+} // namespace detail
+} // namespace ext::oneapi::experimental
+} // namespace _V1
+} // namespace sycl
+
+#else
+
+#include <ctime>
+#include <filesystem>
+#include <fstream>
+#include <random>
+#include <regex>
+#include <sstream>
+
+namespace sycl {
+inline namespace _V1 {
+namespace ext::oneapi::experimental {
+namespace detail {
+
+std::string generate_semi_unique_id() {
```

Comment:
I see none of the functions on this file following LLVM guidelines about function names (camel case starting with lower case). Is there any reason?

Reply:
Same for variable names.

Reply:
done